### ISSUE TITLE
Python 3.12 Support - versioneer currently unsupported in Python 3.12

### ISSUE THREAD

**Issue by FriedLabJHU** [holehouse-lab/protfasta#4]
It looks like python-versioneer is only [experimentally](https://github.com/python-versioneer/python-versioneer?tab=readme-ov-file#the-versioneer) supported in Python 3.12, effectively breaking installation of packages that rely on versioneer for version control; including ```protfasta```.

Should this tool migrate to ```hatch``` to simplify and upgrade package management?

**Comment by alexholehouse** [comment]
Yeah great point - we will be moving all our tools from versioneer over to [versioningit](https://pypi.org/project/versioningit/) over the coming year to maintain compatibility/consistency with the MOLSSI [cookiecutter](https://www.cookiecutter.io/) templates. We have pushed upper Python limits to 3.11 in some places (given that 3.12 came out on Oct 23, this doesn't feel so egregious right now, anyway) but not on Protasta. 

I have just pushed this update, so protfasta now is capped at Python 3.11 (avoids this issue for now) and we have the great versioneer --> versioningit transition on our roadmap going forward. 

I'll keep this issue open until we actually resolve it with the versioningit transition.

**Comment by alexholehouse** [comment]
OK, this took a while (...) but I'm pleased to share **protfasta** is fully transitioned to `pyproject.toml` and `versioningit` bringing support to 3.12 and 3.13 (and beyond). Thanks for your patience here. We are now actively converting our stack over.

Let me know if you run into any issues. 0.1.15.2 (current version) is on PyPI as of now, and we'll be putting it through its paces...  

I'm gonna leave this open for a touch longer incase anything goes awry, but things seem stable/robust...
